changefeedccl: deflake some tests

danhhz opened the pull request:
see individual commits for details

danhhz committed:
changefeedccl: deflake TestManyChangefeedsOneTable

In #34035, we switched from lib/pq to pgx in the sinkless changefeed
tests. That PR fixed a few places where we were accidentally relying on
lib/pq specific behavior, but missed this one. Basically lib/pq waits
until the first result comes back from the server before returning from
a `Query`, but pgx doesn't. This means some of the changefeeds start
later than they seem to, causing test flakes.

Closes #34091
Closes #34078

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/34104)
<!-- Reviewable:end -->

danhhz committed:
changefeedccl: deflake TestChangefeedRetryableSinkError

This channel wasn't getting closed if the stopper rejected the async
task, which casued `(*changeAggregator).close()` to hang.

Release note: None

danhhz committed:
changefeedccl: deflake TestChangefeedSchemaChangeAllowBackfill//add_column

Same deal as always, the `ALTER TABLE ADD COLUMN` could have run before
the feed finished setting up.

Release note: None

ajwerner reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/34104#-:-LWWk9Ma9lOiOgKLSTu3:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @andreimatei)



<!-- Sent from Reviewable.io -->

danhhz commented:
bors r=ajwerner
